crates/chia-consensus/src/test_generators.rs

```diff
@@ -438,6 +461,84 @@ fn run_generator(#[case] name: &str) {
         }
     }
 
+    // compare costed_sha with default and expected file
+    let mut a2 = make_allocator(COST_SHATREE | DONT_VALIDATE_SIGNATURE);
+    let conds_sha = run_block_generator2(
+        &mut a2,
+        &generator,
+        &block_refs,
+        11_000_000_000,
+        COST_SHATREE | DONT_VALIDATE_SIGNATURE,
+        &Signature::default(),
+        None,
+        &TEST_CONSTANTS,
+    );
+
+    let (_expected_cost, sha_output) = match conds_sha {
+        Ok(ref conditions) => {
+            let cond_cost: u64 = conditions.spends.iter().map(|v| v.condition_cost).sum();
+            assert_eq!(cond_cost, conditions.condition_cost);
+            let exe_cost: u64 = conditions.spends.iter().map(|v| v.execution_cost).sum();
+            // the generator itself has execution cost. At least the cost of
+            // a quote
+            assert!(exe_cost <= conditions.execution_cost);
+            (conditions.cost, print_conditions(&a2, &conditions, &a2))
+        }
+        Err(code) => (0, format!("FAILED: {}\n", u32::from(code.1))),
+    };
+
+    // new costing failure
+    if sha_output == "FAILED: 23\n" {
+        assert_eq!(costed_sha_cost, sha_output);
+        write_back.push_str(&format!("COSTED_SHA:\n{sha_output}"));
+    } else if !conds_sha.is_ok() {
+        // otherwise check failures match
+        // assert_eq!(sha_output, last_output);
+    } else {
```

Comment:
it looks like you might be able to have early returns in the above two blocks, and then not have this `else`-block, and de-indent a lot of code below